torch/nn/functional.py

```diff
@@ -3262,3 +3263,147 @@ def _in_proj_v(weight, bias, embed_dim, value):
+                                      norm2_normalized_shape,               # type: Int
+                                      norm2_weight,                         # type: Tensor
+                                      norm2_bias,                           # type: Tensor
+                                      norm2_eps,                            # type: Float
+                                      dropout_p,                            # type: Float
+                                      dropout_training,                     # type: Bool
+                                      dropout_inplace,                      # type: Bool                      
+                                      dropout1_p,                           # type: Float
+                                      dropout1_training,                    # type: Bool
+                                      dropout1_inplace,                     # type: Bool
+                                      dropout2_p,                           # type: Float
+                                      dropout2_training,                    # type: Bool
+                                      dropout2_inplace,                     # type: Bool
+                                      linear1_weight,                       # type: Tensor
+                                      linear1_bias,                         # type: Tensor
+                                      linear2_weight,                       # type: Tensor
+                                      linear2_bias                          # type: Tensor
+                                      ):
+    # output type -> Tensor
+    r"""The forward function of the encoder layer in the Transformer model.
+
+    See :class:`~torch.nn.TransformerEncoderLayer` for details.
+    """
+    src2 = layer_norm(src, norm1_normalized_shape, norm1_weight, norm1_bias, norm1_eps)
+    src2 = multi_head_attention_forward(src2, src2, src2, self_attn_embed_dim, self_attn_num_heads,
+                                        self_attn_in_proj_weight, self_attn_in_proj_bias, self_attn_bias_k,
+                                        self_attn_bias_v, self_attn_add_zero_attn,
+                                        self_attn_dropout, self_attn_out_proj_weight, 
+                                        self_attn_out_proj_bias, attn_mask=src_mask)[0]
+    src2 = dropout(src2, dropout1_p, dropout1_training, dropout1_inplace)
+    src = src + src2
+    src2 = layer_norm(src, norm2_normalized_shape, norm2_weight, norm2_bias, norm2_eps)
+    src2 = linear(src2, linear1_weight, linear1_bias)
+    src2 = relu(src2)
+    src2 = dropout(src2, dropout_p, dropout_training, dropout_inplace)
+    src2 = linear(src2, linear2_weight, linear2_bias)
+    src2 = dropout(src2, dropout2_p, dropout2_training, dropout2_inplace)
+    src = src + src2
+
+    return src
+
+@weak_script
+def transformer_decoder_layer_forward(tgt,                                  # type: Tensor
```

Comment:
Add to the docs that this is an unstable API, because we'll need to add support for slightly different standard transformers (e.g. BERT) and this is based on "All you need is attention" (which we should also cite and reference).